### PR TITLE
update(JS): web/javascript/reference/global_objects/string/matchall

### PR DESCRIPTION
**Pull request by undead404**
Оригінальний вміст: [String.prototype.matchAll()@MDN](https://developer.mozilla.org/en-us/docs/Web/JavaScript/Reference/Global_Objects/String/matchAll), [сирці String.prototype.matchAll()@GitHub](https://github.com/mdn/content/blob/main/files/en-us/web/javascript/reference/global_objects/string/matchall/index.md)

Нові зміни:
- [mdn/content@d85a7ba](https://github.com/mdn/content/commit/d85a7ba8cca98c2f6cf67a0c44f0ffd467532f20)
- [mdn/content@ab97df6](https://github.com/mdn/content/commit/ab97df6ce8865569507bcfc884206a1ed297a690)
- [mdn/content@f3df525](https://github.com/mdn/content/commit/f3df52530f974e26dd3b14f9e8d42061826dea20)
- [mdn/content@2eb202a](https://github.com/mdn/content/commit/2eb202adbe3d83292500ed46344d63fbbae410b5)
- [mdn/content@7b72e7d](https://github.com/mdn/content/commit/7b72e7d4860bcac018713414df3ca7e941cb7bd3)
- [mdn/content@9e52d33](https://github.com/mdn/content/commit/9e52d335675a9f7daa34b5e24c9b1292dbf23b2e)
- [mdn/content@ce29091](https://github.com/mdn/content/commit/ce2909126eb09e44c9f48d9f65d072acae827749)
- [mdn/content@968e6f1](https://github.com/mdn/content/commit/968e6f1f3b6f977a09e116a0ac552459b741eac3)
- [mdn/content@ba66c42](https://github.com/mdn/content/commit/ba66c4221816fc47bd11282058b1e5ff9d447b96)
- [mdn/content@8d76e99](https://github.com/mdn/content/commit/8d76e99363dab6c2f8d31216b5b28620dd1e041f)
- [mdn/content@3fe8522](https://github.com/mdn/content/commit/3fe8522d9432bd9bf068b937292a3f7499b461af)
- [mdn/content@83d09fa](https://github.com/mdn/content/commit/83d09fa55f408e15f68d519c913ec0ee5d126fb7)
- [mdn/content@d6f63f6](https://github.com/mdn/content/commit/d6f63f6e41146f459e21046af1c51fadb772de70)
- [mdn/content@8bf018f](https://github.com/mdn/content/commit/8bf018f0a39d012a0d98afe3f15e0ed0fb7c8ce5)
- [mdn/content@42428ac](https://github.com/mdn/content/commit/42428ac9cd08d082ce124333a0d581d02d76d5bf)
- [mdn/content@35b6d1b](https://github.com/mdn/content/commit/35b6d1b1490ac94c8e58dcf234eda32a5ffc9276)
- [mdn/content@edaa4af](https://github.com/mdn/content/commit/edaa4aff4cf7532efdcba6014b9316aad2a5b6f1)

### CODE REVIEW

**Comment by undead404** [comment]
Оновлено

File: files/uk/web/javascript/reference/global_objects/string/matchall/index.md [hunk 1]
```diff
@@ -1,128 +1,146 @@
...
 }
+// Знайдено слово football, початок=6, закінчення=14.
+// Знайдено слово foosball, початок=16, закінчення=24.
 ```
 
-За наявності методу `matchAll` є можливість уникати циклу {{jsxref("Statements/while", "while")}} і використання методу `exec` із `g`.
-
-Натомість під час роботи з `matchAll` створюється ітератор, який можна використати зі зручнішими конструкціями {{jsxref("Statements/for...of", "for...of")}},
-{{jsxref("Operators/Spread_syntax", "array spread")}}, чи {{jsxref("Array.from()")}}:
+За наявності методу `matchAll` є можливість уникати циклу {{jsxref("Statements/while", "while")}} і використання методу `exec` із `g`. Повертається ітератор, який можна використовувати зі зручнішими конструкціями – {{jsxref("Statements/for...of", "for...of")}}, [розгортанням масиву](/uk/docs/Web/JavaScript/Reference/Operators/Spread_syntax) і {{jsxref("Array.from()")}}:
```
Comment (KlashReik) [inline files/uk/web/javascript/reference/global_objects/string/matchall/index.md:63]:
```suggestion
За наявності методу `matchAll` є можливість уникати циклу {{jsxref("Statements/while", "while")}} і використання методу `exec` із `g`. Натомість повертається ітератор, який можна використовувати зі зручнішими конструкціями – {{jsxref("Statements/for...of", "for...of")}}, [розгортанням масиву](/uk/docs/Web/JavaScript/Reference/Operators/Spread_syntax) і {{jsxref("Array.from()")}}:
```

**Review by KlashReik — APPROVED** [review]

File: files/uk/web/javascript/reference/global_objects/string/matchall/index.md (hunk 1 above)
Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/string/matchall/index.md:63]: По-моєму, "натомість" тут недоречно

Comment (KlashReik) [reply files/uk/web/javascript/reference/global_objects/string/matchall/index.md:63]: А яке слово буде доречним? Без нього досить важко прочитати і зрозуміти зміст. "Замість цього" тобі більше подобається?

Comment (undead404) [reply files/uk/web/javascript/reference/global_objects/string/matchall/index.md:63]: Якщо дуже хочеться, то "Це можливо завдяки тому, що...", Або "Цей метод повертає..."

Comment (KlashReik) [reply files/uk/web/javascript/reference/global_objects/string/matchall/index.md:63]:
```suggestion
За наявності методу `matchAll` є можливість уникати циклу {{jsxref("Statements/while", "while")}} і використання методу `exec` із `g`. Це можливо завдяки тому, що повертається ітератор, який можна використовувати зі зручнішими конструкціями – {{jsxref("Statements/for...of", "for...of")}}, [розгортанням масиву](/uk/docs/Web/JavaScript/Reference/Operators/Spread_syntax) і {{jsxref("Array.from()")}}:
```

**Pull request merged**